src/pages/core/guides/oracle-plugin-example.md

```diff
@@ -0,0 +1,375 @@
+
+### Anchor Overview
+
+In this guide, we'll use the Anchor framework, but you can also implement it using a native program. Learn more about the Anchor framework here.
+
+For simplicity, we'll use a mono-file approach, with helpers, state, accounts, and instructions all in lib.rs instead of the usual separation.
+
+*Note: You can follow along and open the example in Solana Playground, an online tool to build and deploy Solana programs: [Solana Playground](https://solanaplayground.io/).*
+
+### Helpers & Constants
+
+Instead of declaring some inputs repeatedly, it’s a good idea to create constants that we can easily reference in our instructions/functions. 
+
+**Here are the constants used in this oracle protocol:**
+```rust
+// Constants
+const SECONDS_IN_AN_HOUR: i64 = 3600;
+const SECONDS_IN_A_MINUTE: i64 = 60;
+const SECONDS_IN_A_DAY: i64 = 86400;
+
+const MARKET_OPEN_TIME: i64 = 14 * SECONDS_IN_AN_HOUR + 30 * SECONDS_IN_A_MINUTE; // 14:30 UTC == 9:30 EST
+const MARKET_CLOSE_TIME: i64 = 21 * SECONDS_IN_AN_HOUR; // 21:00 UTC == 16:00 EST
+const MARKET_OPEN_CLOSE_MARGIN: i64 = 15 * SECONDS_IN_A_MINUTE; // 15 minutes in seconds
+const REWARD_IN_LAMPORTS: u64 = 10000000; // 0.001 SOL
+```
+
+Creating helpers to check some of the logic of our smart contract makes sense, such as checking if the US market is open and if it’s within 15 minutes of opening or closing.
+
+**is_us_market_open helper:**
+```rust
+fn is_us_market_open(unix_timestamp: i64) -> bool {
+    let seconds_since_midnight = unix_timestamp % SECONDS_IN_A_DAY;
+    let weekday = (unix_timestamp / SECONDS_IN_A_DAY + 4) % 7;
+
+    // Check if it's a weekday (Monday = 0, ..., Friday = 4)
+    if weekday >= 5 {
+        return false;
+    }
+
+    // Check if current time is within market hours
+    seconds_since_midnight >= MARKET_OPEN_TIME && seconds_since_midnight < MARKET_CLOSE_TIME
+}
+```
+This helper checks if the US market is open based on the given Unix timestamp by calculating the seconds since midnight and the day of the week. If the current time is a weekday and is within market hours, it returns true.
+
+**is_within_15_minutes_of_market_open_or_close helper:**
+```rust
+fn is_within_15_minutes_of_market_open_or_close(unix_timestamp: i64) -> bool {
+    let seconds_since_midnight = unix_timestamp % SECONDS_IN_A_DAY;
+
+    // Check if current time is within 15 minutes after market open or within 15 minutes after market close
+    (seconds_since_midnight >= MARKET_OPEN_TIME && seconds_since_midnight < MARKET_OPEN_TIME + MARKET_OPEN_CLOSE_MARGIN) ||
+    (seconds_since_midnight >= MARKET_CLOSE_TIME && seconds_since_midnight < MARKET_CLOSE_TIME + MARKET_OPEN_CLOSE_MARGIN)
+}
+```
+
+This helper checks if the current time is within 15 minutes of the market opening or closing by calculating the seconds since midnight and comparing it with the market open and close times, adding a 15-minute margin.
```

Comment:
Sounds quite similar to line 102?

Reply:
Yes because at the end of the day they do the same stuff but one has the 15m margin check in addition.